File: dotnet/cafdemalihapa/Directories/Repository/Dotnet/Cafdemalihapa/Scripts/EnvironmentVariablesSource/CodeBases/CodeBaseDirectory.cs
using cafdemalihapa.Applications.Cafdemalihapa;

namespace cafdemalihapa.Directories.Repository.Dotnet.Cafdemalihapa.Scripts.EnvironmentVariablesSource.CodeBases
{
    public class CodeBaseDirectory(
        EnvironmentVariablesSourceDirectory environmentVariablesSourceDirectory,
        CodeBase codeBase
    )
    {
        public string GetPath()
        {
            var codeBaseValue = codeBase.GetCodeBaseValue();
            return Path.Combine(environmentVariablesSourceDirectory.GetPath(), codeBaseValue);
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Repository/Dotnet/Cafdemalihapa/Scripts/EnvironmentVariablesSource/CodeBases/CodeBaseDirectory.fs
namespace cafdemwimu.console.Directories.Repository.Dotnet.Cafdemalihapa.Scripts.EnvironmentVariablesSource.CodeBases

open System.IO
open cafdemwimu.console.Applications.Cafdemalihapa
open cafdemwimu.console.Directories.Repository.Dotnet.Cafdemalihapa.Scripts.EnvironmentVariablesSource

type CodeBaseDirectory(environmentVariablesSourceDirectory: EnvironmentVariablesSourceDirectory, codeBase: CodeBase) =
    member _.GetPath() =
        let codeBaseValue = codeBase.GetCodeBaseValue()
        Path.Combine(environmentVariablesSourceDirectory.GetPath(), codeBaseValue)
